travis/build_total.py

```diff
@@ -0,0 +1,180 @@
+        self.clone_merge_repository('libecl', self.ecl_pr_num, basedir)
+        self.clone_merge_repository('libres', self.res_pr_num, basedir)
+        self.clone_merge_repository('ert'   , self.ert_pr_num, basedir)
+
+    def clone_merge_repository(self, rep_name, pr_num, basedir):
+        if (self.rep_name != rep_name):
+            os.chdir(basedir)
+            subprocess.check_call(["git", "clone", "https://github.com/Statoil/%s" % rep_name])
+            if (pr_num >= 0):
+                rep_path = os.path.join(basedir, rep_name)
+                cwd = os.getcwd()
+                os.chdir(rep_path)
+                s = "refs/pull/%d/head:x" % pr_num
+                subprocess.check_call(["git", "config", "user.email", "you@example.com"])
+                subprocess.check_call(["git", "config", "user.name", "Your Name"])
+                subprocess.check_call(["git", "fetch", "https://github.com/Statoil/%s" % rep_name, s])
+                subprocess.check_call(["git", "merge", "x"])
+                os.chdir(cwd) 
+            
+    def compile_and_build(self, basedir):
+        install_dir = os.path.join(basedir, "install")
+        if not os.path.isdir(install_dir):
+            os.makedirs(install_dir)
+        self.compile_ecl(basedir, install_dir)
+        self.compile_res(basedir, install_dir)
+        self.compile_ert(basedir, install_dir)
+
+    def build(self, source_dir, install_dir, test):
+        build_dir = os.path.join(source_dir, "build")
+        if not os.path.isdir(build_dir):
+            os.makedirs(build_dir)
+        cmake_args = ["cmake", source_dir, "-DBUILD_TESTS=ON", "-DBUILD_PYTHON=ON", "-DERT_BUILD_CXX=ON", "-DBUILD_APPLICATIONS=ON", "-DCMAKE_INSTALL_PREFIX=%s" % install_dir, 
+                      "-DINSTALL_ERT_LEGACY=ON", "-DCMAKE_PREFIX_PATH=%s" % install_dir, "-DCMAKE_MODULE_PATH=%s/share/cmake/Modules" % install_dir]
+        cwd = os.getcwd()
+        os.chdir(build_dir)
+        subprocess.check_call(cmake_args)
+        subprocess.check_call(["make"])
+        if test:
+            subprocess.check_call(["ctest"])
+        subprocess.check_call(["make" , "install"])
+        subprocess.check_call(["bin/test_install"])
+        os.chdir(cwd) 
+
+    def compile_ecl(self, basedir, install_dir):
+        source_dir = os.path.join(basedir, "libecl")
+        if (self.repository == 'ecl'):
+            test = True
+        else:
+            test = False
+        self.build(source_dir, install_dir, test)
+
+    def compile_res(self, basedir, install_dir):
+        source_dir = os.path.join(basedir, "libres")
+        if (self.repository == 'ecl' or self.repository == 'res'):
+            test = True
+        else:
+            test = False
+        self.build(source_dir, install_dir, test)
+
+    def compile_ert(self, basedir, install_dir):
+        source_dir = os.path.join(basedir, "ert")
+        self.build(source_dir, install_dir, True)
+        
+
+def main():
+    
+    os.chdir("..")
```

Comment:
I fear that stepping out of the directory *might* lead to conflicts between different builds; I suggest you just remove the `os.chdir( )` and let the rest of the code be as-is. Then the other repos, including the `install/` folder will be built inside the current project - that is fine.